Sprint-1/step-2

Neverhood93 opened the pull request:
Создан проект по шаблону CRA --template typescript.
Настроена структура папок проекта.
Сверстана главная страница.
Настроено получение данных ингредиентов из API.
Добавлены модальные окна и вызов их.

Neverhood93 committed:
Сформирована структура папок для проекта.
Создан черновик для AppHeader.

Neverhood93 committed:
Новая версия header.

Neverhood93 committed:
Финал верстки header.

Neverhood93 committed:
Черновик для BurgerIngredients.

Neverhood93 committed:
Финал BurgerIngredients.

Neverhood93 committed:
Добавлен BurgerConstructorList.

Neverhood93 committed:
Добавлен компонент TotalPrice и кнопка с отправкой заказа.

Neverhood93 committed:
Добавлено игнорирование папки ".idea"

Neverhood93 committed:
Исправление замечаний от ревьюера.

Neverhood93 committed:
Старые компоненты переделаны на tsx.
Добавлены интерфейсы, которые вынесены в отдельный файл.
Реализовано получение ингредиентов из API.
Удалены захардкоженные данные data.js.
Добавлен Prettier.

Neverhood93 committed:
Добавлено базовое модальное окно Modal и ModalOverlay.
Добавлены черновые компоненты IngredientDetails и OrderDetails.
Изменен модуль App для вызова модальных окон.

Neverhood93 committed:
Доработана верстка IngredientDetails и OrderDetails.

kyzinatra commented on src/components/app/app.tsx:
```diff
@@ -0,0 +1,94 @@
+    try {
+      const response = await fetch(API_URL);
+      const data = await response.json();
+      setIngredients(data.data);
```
Надо исправить: Стоит добавить проверку успешности выполнения с помощью `res.ok` так как `catch` ловит только ошибки в самом запросе, но если сервер вернет к примеру 400, то ее `catch` не поймает.

[Подробнее:](https://developer.mozilla.org/ru/docs/Web/API/Fetch_API/Using_Fetch) (https://developer.mozilla.org/ru/docs/Web/API/Fetch_API/Using_Fetch)

kyzinatra commented on src/components/app/app.tsx:
```diff
@@ -0,0 +1,94 @@
+    setLoading(true);
+    try {
+      const response = await fetch(API_URL);
+      const data = await response.json();
```
Можно лучше: все проверки и преобразования стоит вынести в отдельную функцию. Тут только вызвать функциюю `getIngredients().then(r => setIngredients(r.data))`

kyzinatra commented on src/components/app/app.tsx:
```diff
@@ -0,0 +1,94 @@
+
+      {isIngredientModalOpen && selectedIngredient && (
+        <Modal title="Детали ингредиента" onClose={closeIngredientModal}>
+          <IngredientDetails ingredient={selectedIngredient} />
```
Можно лучше: Модальные окна имеют внутри порталы. Их можно писать где угодно. Удобно сделать это рядом с кнопкой вызова модального окна. Так не придется прокидывать лишние функции в пропсы

kyzinatra commented on src/components/modal/modal.tsx:
```diff
@@ -0,0 +1,45 @@
+  title: string;
+  onClose: () => void;
+  children: React.ReactNode;
+}
```
Можно лучше: типизировать `children` можно как `PropsWithChildren`

kyzinatra commented on src/components/modal/modal.tsx:
```diff
@@ -0,0 +1,45 @@
+    <>
+      <ModalOverlay onClose={onClose} />
+      <div className={styles.modal}>
+        <div className={styles.header}>
```
Отлично! Отличная релизация модальных окон

kyzinatra commented:
Выглядит очень круто! Видно, что проделана большая работа. Но все же есть моменты на которые стоит обратить внимание.
Подробнее: 

**Что важно исправить:**

- Стоит добавить проверку успешности выполнения с помощью `res.ok` так как `catch` ловит только ошибки в самом запросе, но если сервер вернет к примеру 400, то ее `catch` не поймает.

[Подробнее:](https://developer.mozilla.org/ru/docs/Web/API/Fetch_API/Using_Fetch) (https://developer.mozilla.org/ru/docs/Web/API/Fetch_API/Using_Fetch)

**Можно лучше:**

-  все проверки и преобразования стоит вынести в отдельную функцию. Тут только вызвать функциюю `getIngredients().then(r => setIngredients(r.data))`
-  Модальные окна имеют внутри порталы. Их можно писать где угодно. Удобно сделать это рядом с кнопкой вызова модального окна. Так не придется прокидывать лишние функции в пропсы
-  типизировать `children` можно как `PropsWithChildren`


**Что понравилось:**

-  Отличная релизация модальных окон


**Обратите внимание, что работа принимается только после исправления всех `«Надо исправить»`.**

Neverhood93 committed:
Исправлено замечание:
Стоит добавить проверку успешности выполнения с помощью res.ok так как catch ловит только ошибки в самом запросе, но если сервер вернет к примеру 400, то ее catch не поймает.

Neverhood93 committed:
Исправлено замечание:
Все проверки и преобразования стоит вынести в отдельную функцию. Тут только вызвать функциюю getIngredients().then(r => setIngredients(r.data)).

Neverhood93 committed:
Исправлено замечание:
Модальные окна имеют внутри порталы. Их можно писать где угодно. Удобно сделать это рядом с кнопкой вызова модального окна. Так не придется прокидывать лишние функции в пропсы.

Neverhood93 committed:
Исправлено замечание:
Типизировать children можно как PropsWithChildren для модуля Modal.

kyzinatra commented:
Вы отлично потрудились.
Поздравляю! Ваша работа принята.
Удачного дальнейшего обучения.
